web/src/compose_recipient.ts

```diff
@@ -331,6 +331,19 @@ export function initialize(): void {
     $("#private_message_recipient").on("input", restore_placeholder_in_firefox_for_no_input);
 }
 
+export function update_topic_inputbox_on_mandatory_topics_change(): void {
+    if (realm.realm_mandatory_topics) {
+        const $input = $("input#stream_message_recipient_topic");
+        $input.attr("placeholder", "Topic");
+        $input.removeClass("empty-topic-display");
+        const $topic_not_mandatory_placeholder = $("#topic-not-mandatory-placeholder");
+        $topic_not_mandatory_placeholder.css({visibility: "hidden"});
+        $topic_not_mandatory_placeholder.hide();
+        return;
+    }
+    update_topic_displayed_text(compose_state.topic());
```

Comment:
Hmm, does this undo all the things the previous case did?

It'd be a lot easier to verify if this function just toggled some CSS classes, and didn't also do `.css(` calls.

Reply:
`update_topic_displayed_text()` is idempotent for `realm.realm_mandatory_topics=false`. It just resets everything and applies the needed css based on the state of compose box to show the right UI.

We can include the code block for `realm_mandatory_topics=true` defined in `update_topic_inputbox_on_mandatory_topics_change` to `update_topic_displayed_text` and just simply call it without introducing this new function. 

But we don't need that part to be executed again & again once `realm_mandatory_topics=true` is set (which'd be 99% of the time `update_topic_displayed_text` gets called). It won't cause any bug, but is of no benefit. OTOH, `//reset` code in `update_topic_displayed_text` always needs to get executed when `realm_mandatory_topics=false`

Reply:
Added a prep commit to avoid using `.css()`.